BHoM_Adapter/AdapterActions/Pull.cs

```diff
@@ -42,6 +42,33 @@ public abstract partial class BHoMAdapter
         /* These methods represent Actions that the Adapter can complete. 
            They are publicly available in the UI as individual components, e.g. in Grasshopper, under BHoM/Adapters tab. */
 
+        [Description("Performs a set up for the Request input of the Pull Action.")]
+        public virtual bool SetupPullRequest(object request, out IRequest actualRequest)
+        {
+            // Always assure there is a Request. Allow to input a Type to generate a FilterRequest.
+            actualRequest = null;
+
+            if (request is Type)
+                actualRequest = BH.Engine.Data.Create.FilterRequest((Type)request, "");
+
+            if (typeof(IRequest).IsAssignableFrom(request.GetType()))
+                actualRequest = request as IRequest;
+
+            if (request == null)
+                actualRequest = new FilterRequest();
+
+            return true;
```

Comment:
What happens if I provide a Point? Think this method should raise a warning and return false if non of the above cases is true, and that the method for those cases should return false.

Reply:
EDIT: If a message is raised elsewhere, or by the UI, simply returning false if it fails is enough I think.